tests/network/vmi_multus.go

```diff
@@ -1041,7 +1042,19 @@ var _ = Describe("[Serial]SRIOV", func() {
 				migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
 				tests.ConfirmVMIPostMigration(virtClient, vmi, migrationUID)
 
-				Expect(checkMacAddress(vmi, interfaceName, mac)).To(Succeed(),
+				var interfaceName string
+
+				// It may take some time for the VMI interface status to be updated with the information reported by
+				// the guest-agent.
+				Eventually(func() error {
```

Comment:
After looking a little bit, I think this `Eventually` block's lambda can be exactly the same as the one defined on the test setup. 

Couldn't you define it one, and re-use it ?

Reply:
Your comments are contradicting here, later you ask to make some stuff local.

As there are inner asserts here, I do not want to stack them up in helper functions. I guess we can refactor it using a poller helper instead of the `Eventually` later on when the migration tests are working.

But to the point here, except the assertions, we also have some little differences like the additional assert about the interface/mac check in the guest.

Reply:
> Your comments are contradicting here, later you ask to make some stuff local.

Why are they contradicting ? I am just asking for a (single) function (not 2 lambdas) with a locally scoped VMI and interfaceName that checks in the mac address did trickle down to the guest. 

Specifically, I am just saying I don't find 

```golang
    Eventually(func() error {
        var err error
        vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
        Expect(err).NotTo(HaveOccurred())
        interfaceName, err = getInterfaceNameByMAC(vmi, mac)
        return err
    }, 30*time.Second, 5*time.Second).Should(Succeed())

    Expect(checkMacAddress(vmi, interfaceName, mac)).To(Succeed(), "SR-IOV VF is expected to exist in the guest")
```

different to 

```golang
    Eventually(func() error {
        updatedVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
        Expect(err).NotTo(HaveOccurred())
        interfaceName, err := getInterfaceNameByMAC(updatedVMI, mac)
        if err != nil {
            return err
        }
        return checkMacAddress(updatedVMI, interfaceName, mac)
    }, 30*time.Second, 5*time.Second).Should(Succeed(), "SR-IOV VF is expected to exist in the guest after migration")
```

They both check that the MAC appears in the VMI guest os. Did I miss something ? 

> 
> As there are inner asserts here, I do not want to stack them up in helper functions. I guess we can refactor it using a poller helper instead of the `Eventually` later on when the migration tests are working.
> 
> But to the point here, except the assertions, we also have some little differences like the additional assert about the interface/mac check in the guest.

I think you do that assert outside of the eventually - I think the only difference is that you assert inside / outside the eventually block.

Reply:
The first does need to update the `vmi`, it is a blocking implementation and not really a classic assert.
The second is really an assert, just due to the last part (`checkMacAddress`).

So although we could refactor it and partition it to use some shared code, it is not trivial and goes beyond the scope of the fix I want to do here.

If you have something simple to suggest, maybe you can try to suggest the alternative.
And try to avoid stacking the `Except`, you know I do not like it.

Reply:
> So although we could refactor it and partition it to use some shared code, it is not trivial and goes beyond the scope of the fix I want to do here.

Fair enough. Can't say I like the side effect within that `Eventually` block, but what's done is done.